### PR TITLE
Create 200NumberOfIslands.cpp

### PR DESCRIPTION
**Pull request by ShadyD45**

### CODE REVIEW

**Comment by DevMayurJ** [comment]
Line 1034: Char 34: runtime error: addition of unsigned offset to 0x608000000020 overflowed to 0x608000000008 (stl_vector.h)
SUMMARY: UndefinedBehaviorSanitizer: undefined-behavior /usr/bin/../lib/gcc/x86_64-linux-gnu/9/../../../../include/c++/9/bits/stl_vector.h:1043:34

File: LeetcodeProblemsandSolutions/200NumberOfIslands.cpp [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+				if(j<0 || j >=vctGrid[i].size() || i < 0 || i >=vctGrid.size() || vctGrid[i][j] == '0')
+					return;
+				
+				vctGrid[i][j] = '0';	// Make the '1' we found '0'  
+				
+				// Then check for other '1's.	
+				CheckForIsland(vctGrid, i + 1, j);	// Down
+				CheckForIsland(vctGrid, i - 1, j);	// Up
+				CheckForIsland(vctGrid, i, j - 1);	// Left
+				CheckForIsland(vctGrid, i, j + 1);	// Right
```
Comment (DevMayurJ) [inline LeetcodeProblemsandSolutions/200NumberOfIslands.cpp:61]: =============== Leetcode compiler is giving me below error to this line=============
Line 1034: Char 34: runtime error: addition of unsigned offset to 0x608000000020 overflowed to 0x608000000008 (stl_vector.h)
SUMMARY: UndefinedBehaviorSanitizer: undefined-behavior /usr/bin/../lib/gcc/x86_64-linux-gnu/9/../../../../include/c++/9/bits/stl_vector.h:1043:34

Comment (ShadyD45) [reply LeetcodeProblemsandSolutions/200NumberOfIslands.cpp:61]: I think the if condition above causes this, All  conditions for 'i' should be checked before 'j'.

Comment (DevMayurJ) [reply LeetcodeProblemsandSolutions/200NumberOfIslands.cpp:61]: Yes that worked moving 'i' conditions ahead. I will update the code.

**Review by DevMayurJ — APPROVED** [review]

**Pull request merged**